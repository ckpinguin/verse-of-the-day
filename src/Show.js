import React from 'react';
import PropTypes from 'prop-types';

/*
Conditionally showing children componenents (a simpler variant of ToggleDisplay)

Usage example:
<Show if={this.props.reports.hasNextPage()}>
    <LoadMore onClick={this.handleLoadMore} />
</Show>
*/

Show.propTypes = {
    if: PropTypes.bool
}

export default function Show(props) {
    if (props.if === false) {
        return null;
    }

    let style = {};
    // style.display = 'none';
    
    return (
            <div style={style} >
                {props.children}
            </div>
    );
}