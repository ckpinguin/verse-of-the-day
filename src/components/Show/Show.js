import React from 'react';
import PropTypes from 'prop-types';

/*
Conditionally showing children componenents (a simpler variant of ToggleDisplay)

Usage example:
<Show if={this.props.reports.hasNextPage()}>
    <LoadMore onClick={this.handleLoadMore} />
</Show>
*/
Show.propTypes = {
    if: PropTypes.bool
};
function Show(props) {
    if (props.if === false) {
        return null;
    }

    const style = {};
    // style.display = 'none';
    
    return (
        React.cloneElement(props.children, { style: {style} })
    );
}
export default Show;
