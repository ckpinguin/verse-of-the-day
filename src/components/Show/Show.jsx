import React from 'react';
import PropTypes from 'prop-types';

/*
Conditionally showing children componenents (a simpler variant of ToggleDisplay)

Usage example:
<Show if={this.props.reports.hasNextPage()}>
    <LoadMore onClick={this.handleLoadMore} />
</Show>
*/
Show.propTypes = {
    if:         PropTypes.bool,
    children:   PropTypes.any.isRequired
};
Show.defaultProps = {
    if: true
};
function Show(props) {
    if (props.if === false) {
        return null;
    }

    const style = {};
    // style.display = 'none';

    return (
        <div>
            {React.Children.map(props.children, (child, i) => {
                return React.cloneElement(child, { style: {style} });
            })}
        </div>
    )
    
}
export default Show;
