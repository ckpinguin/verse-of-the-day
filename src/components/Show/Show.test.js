import React from 'react';
import ReactDOM from 'react-dom';

import Show from './Show';

describe('<Show />', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Show />, div);
    });
});