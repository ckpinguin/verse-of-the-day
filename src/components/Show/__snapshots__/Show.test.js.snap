// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Show /> did not change UI compared to recorded snapshot 1`] = `
<div>
  <div
    id="showme"
    style={Object {}}
  />
</div>
`;
